lib/generators/metagem/engine_model/engine_model_generator.rb

```diff
@@ -0,0 +1,59 @@
+require_relative '../config/engine_config'
+
+class Metagem::EngineModelGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path('templates', __dir__)
+
+  def move_migration
+    return if config.project_last_migration_file_name.blank?
+
+    copy_file(
+      config.project_last_migration_path,
+      config.gem_migrations_path(config.project_last_migration_file_name)
+    )
+    remove_file(config.project_last_migration_path)
+  end
+
+  def move_model
+    model_path = config.project_models_path("#{name}.rb")
+    new_model_path = config.gem_models_path("#{name}.rb")
+    copy_file(model_path, new_model_path, force: true)
+    remove_dir(config.project_models_path(config.gem_name), force: true)
+
+    gsub_file(new_model_path, "ApplicationRecord") do |_match|
+      "#{config.gem_class}::ApplicationRecord"
+    end
+  rescue Thor::Error => e
+    puts e.message
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.